tests/PHPStan/Analyser/data/call-user-func-array.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace CallUserFuncArray;
+
+use function PHPStan\Testing\assertType;
+
+/**
+ * @template T
+ * @param T $t
+ * @return T
+ */
+function generic($t) {
+	return $t;
+}
+
+function fun(): int
+{
+	return 3;
+}
+
+function fun3($i, $x, $y): int
+{
+	return 3;
+}
+
+class c {
+	static function m(): string
+	{
+		return 'hello';
+	}
+}
+
+class Foo {
+	function proxy() {
+		$params = [
+			'CallUserFuncArray\generic',
+			[123]
+		];
+
+		assertType('mixed', call_user_func_array(...$params));
+
+		$params = [
+			'CallUserFuncArray\generic',
+			123,
+			456
+		];
+
+		// cufa expect max 2 args
+		assertType('*NEVER*', call_user_func_array(...$params));
+		assertType('mixed', call_user_func(...$params));
+	}
+
+	/**
+	 * @param string $params,...
+	 */
+	function doVariadics(...$params) {
+		assertType('string', call_user_func_array('CallUserFuncArray\generic', $params));
+		assertType('string', call_user_func('CallUserFuncArray\generic', ...$params));
+		assertType('array<int|string, string>', call_user_func('CallUserFuncArray\generic', $params));
```

Comment:
I have no idea why php7.x returns `'array<int, string>'` while php 8.x a `'array<int|string, string>'`

Reply:
at least the diff between php 7.x and 8.x is reproducible on https://phpstan.org/r/58c30346-9568-47ca-82e5-53b2fffda7d0

Reply:
Because of named arguments. https://3v4l.org/HqRop

Reply:
thx for the insight. just added a separate test for php7/php8